### PR TITLE
Add quick-start page

### PR DESCRIPTION
**Pull request by marcobottaro**
#### List of Changes
Add API tester part component

#### Motivation and Context
It was missing

#### How Has This Been Tested?
Manually

##### Desktop
image
image

##### Mobile
image
image

#### Types of changes

- [ ] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: apps/nextjs-website/src/components/organisms/TabSwitcherProvider/TabSwitcherProvider.tsx [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+  const { activeTabID, setActiveTabID } = context;
+
+  const isActive = id === activeTabID;
+
+  return (
+    <Box
+      onClick={() => setActiveTabID(id)}
+      sx={{
+        backgroundColor: isActive ? '#e6eef9' : 'transparent',
```
Comment (datalek) [inline apps/nextjs-website/src/components/organisms/TabSwitcherProvider/TabSwitcherProvider.tsx:41]: Should we take values like `#e6eef9` from theme?

File: package.json [hunk 2]
```diff
@@ -15,5 +15,9 @@
   "devDependencies": {
     "husky": "^8.0.3",
     "turbo": "^1.9.6"
+  },
+  "dependencies": {
+    "@types/react-syntax-highlighter": "^15.5.7",
+    "react-syntax-highlighter": "^15.5.0"
```
Comment (datalek) [inline package.json:21]: We should move these dependencies into `nextjs-website` module. Moreover the `@type` should be installed as `devDependecies`

**Review by datalek — CHANGES_REQUESTED** [review]

File: apps/nextjs-website/src/components/organisms/ApiTesterPart/ApiTesterPart.tsx [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+          </Button>
+        )}
+      </Box>
+      <CodeBlockPart
+        code={isLifeCycleCallPhase ? apiCall.code : apiResponse.code}
+        language={
+          isLifeCycleCallPhase ? apiCall.language : apiResponse.language
+        }
+        mode='dark'
+        title={isLifeCycleCallPhase ? 'Chiamata' : 'Risposta'}
```
Comment (datalek) [inline apps/nextjs-website/src/components/organisms/ApiTesterPart/ApiTesterPart.tsx:80]: What about moving `Chiamata` and `Risposta` into translations?

File: apps/nextjs-website/src/components/organisms/TabSwitcherProvider/TabSwitcherProvider.tsx [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+  children: ReactNode;
+};
+
+function Tab({ id, children }: TabProps) {
+  const context = useContext(TabSwitcherContext);
+  const { palette } = useTheme();
+  if (!context) {
+    console.error(
+      'TabSwitcherContext not found, did you forget to use TabSwitcherContextProvider?'
+    );
```
Comment (datalek) [inline apps/nextjs-website/src/components/organisms/TabSwitcherProvider/TabSwitcherProvider.tsx:30]: Is this `console.error` needed?

File: apps/nextjs-website/src/components/organisms/TabSwitcherProvider/TabSwitcherProvider.tsx [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+  whenActive: string;
+  children: ReactNode;
+};
+
+function TabPanel({ whenActive, children }: TabPanelProps) {
+  const context = useContext(TabSwitcherContext);
+  if (!context) {
+    console.error(
+      'TabSwitcherContext not found, did you forget to use TabSwitcherContextProvider?'
+    );
```
Comment (datalek) [inline apps/nextjs-website/src/components/organisms/TabSwitcherProvider/TabSwitcherProvider.tsx:66]: Same for this one. Do we need this `console.error`?

File: apps/nextjs-website/src/_contents/appIo/quickStartGuide.ts
Comment (datalek) [inline apps/nextjs-website/src/_contents/appIo/quickStartGuide.ts:1]: Before merging to main we have to check text of this quick-start.

File: apps/nextjs-website/src/components/organisms/TabSwitcherProvider/TabSwitcherProvider.tsx (hunk 4 above)
Comment (MarcoPonchia) [reply apps/nextjs-website/src/components/organisms/TabSwitcherProvider/TabSwitcherProvider.tsx:30]: I think is useful to have a specific error during development. By default React does not show a comprehensive error when component's provider is missing.

File: apps/nextjs-website/src/components/organisms/TabSwitcherProvider/TabSwitcherProvider.tsx (hunk 5 above)
Comment (MarcoPonchia) [reply apps/nextjs-website/src/components/organisms/TabSwitcherProvider/TabSwitcherProvider.tsx:66]: I think is useful to have a specific error during development. By default React does not show a comprehensive error when component's provider is missing.

File: apps/nextjs-website/src/components/molecules/GoToStep/GoToStep.tsx
[the remark this answers is not in the document]
Comment (MarcoPonchia) [reply apps/nextjs-website/src/components/molecules/GoToStep/GoToStep.tsx:1]: No i moved it to `atoms` folder

File: apps/nextjs-website/src/components/organisms/TabSwitcherProvider/TabSwitcherProvider.tsx (hunk 4 above)
Comment (datalek) [reply apps/nextjs-website/src/components/organisms/TabSwitcherProvider/TabSwitcherProvider.tsx:30]: What about using an `Error` instead?

File: apps/nextjs-website/src/components/molecules/QuickStartGuideStepper/QuickStartGuideStepper.tsx [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+                    {index + 1 < steps?.length ? (
+                      <Tab key={index} id={steps[index + 1].anchor}>
+                        <GoToStep
+                          previousOrNext='next'
+                          title={`${steps[index + 1].anchor} - ${
+                            steps[index + 1].title
+                          }`}
+                        />
+                      </Tab>
+                    ) : null}
```
Comment (datalek) [inline apps/nextjs-website/src/components/molecules/QuickStartGuideStepper/QuickStartGuideStepper.tsx:121]:
```suggestion
                    {index - 1 >= 0 && (
                      <Tab key={index} id={steps[index - 1].anchor}>
                        <GoToStep
                          previousOrNext='previous'
                          title={`${steps[index - 1].anchor} - ${
                            steps[index - 1].title
                          }`}
                        />
                      </Tab>
                    )}
                    {index + 1 < steps?.length && (
                      <Tab key={index} id={steps[index + 1].anchor}>
                        <GoToStep
                          previousOrNext='next'
                          title={`${steps[index + 1].anchor} - ${
                            steps[index + 1].title
                          }`}
                        />
                      </Tab>
                    )}
```

File: apps/nextjs-website/src/components/organisms/ApiTesterPart/ApiTesterPart.tsx [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+export type ApiTesterPartProps = {
+  apiRequest: ApiPhaseDescription;
+  apiResponse: ApiPhaseDescription;
+};
+
+const ApiTesterPart = ({ apiRequest, apiResponse }: ApiTesterPartProps) => {
+  const { spacing, palette } = useTheme();
+  const [isLifeCycleCallPhase, setIsLifeCycleCallPhase] = useState(true);
+  const { quickStartGuide } = translations;
+  const boxBorder = `1px solid #DCDCDC`;
```
Comment (datalek) [inline apps/nextjs-website/src/components/organisms/ApiTesterPart/ApiTesterPart.tsx:25]: What do you think to take the color `#DCDCDC` from the theme?

**Review by kin0992 — CHANGES_REQUESTED** [review]

**Review by kin0992 — APPROVED** [review]

**Pull request merged**